test/gst-vaapi/vpp/vpp.py

```diff
@@ -0,0 +1,132 @@
+###
+### Copyright (C) 2018-2019 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ..util import *
+
+@slash.requires(have_gst)
+@slash.requires(*have_gst_element("vaapi"))
+@slash.requires(*have_gst_element("vaapipostproc"))
+@slash.requires(*have_gst_element("checksumsink2"))
+class VppTest(slash.Test):
+  def before(self):
+    self.refctx = []
+
+  def gen_input_opts(self):
+    if self.vpp_element not in ["deinterlace"]:
+      opts =  "filesrc location={source} num-buffers={frames}"
+      opts += " ! rawvideoparse format={mformat} width={width} height={height}"
+    else:
+      opts =  "filesrc location={source}"
+      opts += " ! {gstdecoder}"
+    if self.vpp_element not in ["csc", "deinterlace"]:
+      opts += " ! videoconvert ! video/x-raw,format={hwup_format}"
+
+    return opts
+
+  def gen_output_opts(self):
+    opts = "vaapipostproc"
+    if self.vpp_element not in ["csc"]:
+      if self.vpp_element not in ["deinterlace"]:
+        opts += " format={mformat}"
+      if self.vpp_element in ["scale"]:
+        opts += " width={scale_width} height={scale_height}"
+      else:
+        opts += " width={width} height={height}"
+
+      if self.vpp_element in ["contrast"]:
+        opts += " contrast={mlevel}"
+      elif self.vpp_element in ["saturation"]:
+        opts += " saturation={mlevel}"
+      elif self.vpp_element in ["hue"]:
+        opts += " hue={mlevel}"
+      elif self.vpp_element in ["brightness"]:
+        opts += " brightness={mlevel}"
+      elif self.vpp_element in ["denoise"]:
+        opts += " denoise={mlevel}"
+      elif self.vpp_element in ["sharpen"]:
+        opts += " sharpen={mlevel}"
+      elif self.vpp_element in ["deinterlace"]:
+        opts += " deinterlace-mode=1 deinterlace-method={mmethod}"
+      elif self.vpp_element in ["mirroring"]:
+        opts += " video-direction={mmethod}"
+    else:
+        opts += " ! video/x-raw,format={mcscu}"
+
+    if self.vpp_element in ["deinterlace"]:
+      opts += " ! videoconvert ! video/x-raw,format={mformatu}"
+      opts += " ! checksumsink2 file-checksum=false qos=false frame-checksum=false"
+    else:
+      opts += " ! checksumsink2 file-checksum=false frame-checksum=false"
```

Comment:
It should be ok to add `qos=false` to all cases.  Thus we can eliminate this conditional.

Reply:
will fix it